internal/controller/istio_extension_reconciler.go

```diff
@@ -285,7 +294,31 @@ func (r *IstioExtensionReconciler) buildWasmConfigs(ctx context.Context, state *
 			continue
 		}
 
-		wasmActionSetsForPath, err := wasm.BuildActionSetsForPath(pathID, path, actions)
+		validator, err := validatorBuilder.Build()
+		if err != nil {
+			return nil, fmt.Errorf("failed to build validator for path %s: %w", pathID, err)
+		}
+		var validatedActions []wasm.Action
+		celValidationIssues := celvalidator.NewIssueCollection()
+
+		for _, action := range actions {
+			if err := celvalidator.ValidateWasmAction(action, validator); err != nil {
+				logger.V(1).Info("WASM action is invalid", "action", action, "path", pathID, "error", err)
+				celValidationIssues.Add(celvalidator.NewIssue(action, pathID, err))
+			} else {
+				validatedActions = append(validatedActions, action)
+			}
+		}
+
+		if !celValidationIssues.IsEmpty() {
+			state.Store(celvalidator.StateCELValidationErrors, celValidationIssues)
```

Comment:
I just noticed this line here (and the initialisation of the `IssueCollection` happen within the loop over the paths, effectively overwriting the issues with those from the last path only, I'll shift these bits outside the loop